crates/solver/src/driver/solver_settlements.rs

```diff
@@ -147,6 +147,17 @@ pub fn retain_mature_settlements(
 
     let valid_settlement_indices = find_mature_settlements(min_order_age, &settlements);
 
+    for (_, (solver, settlement)) in settlements
+        .iter()
+        .enumerate()
+        .filter(|(i, _)| !valid_settlement_indices.contains(i))
+    {
+        tracing::debug!(
+            solver_name = %solver.name(), ?settlement,
+            "filtered settlement for not being mature",
```

Comment:
This one is super weird/complex. Not sure how helpful that message would be for external people.
Maybe we can log a link to the PR implementing this check? ;)
Not sure how to best communicate what this check is doing. 🤷‍♂️

Reply:
I thought this was mostly for us to know if an external solver settlement gets filtered because all orders are "immature". We are being asked a lot "why is my settlement not being considered?". The idea with this PR is to mostly be able to answer that with a quick log query:
```
log: "auction{id=...}" and log: "solver_name=..."
```

One thing we can do with this message is make it more explicit as to what it means. I had used "for not being mature" to be as succinct as possible.

Reply:
I thought the purpose for the improved logging was specifically for people testing with a local drive which would then have access to these logs. At least this is what it said in the slack [thread](https://cowservices.slack.com/archives/C0375NV72SC/p1652955145717999) which initiated this work.
I think the wording of this log is as good as it gets without having a wall of text there, though. :/

Reply:
Ah you are right. I misread the Slack post. I'll try to reword it to be more descriptive.